### ISSUE TITLE
Workflow assumes that the names of binaries are the same as their crate.

### ISSUE THREAD

**Issue by olson-sean-k** [rust-build/rust-build.action#5]
The `build.sh` script assumes that the name of the output binary of a build is prefixed with the name of its crate. In particular, [this `find` command](https://github.com/rust-build/rust-build.action/blob/41f6d0dc3dc3be6dee0f2ef48cc55c7e44aed3fb/build.sh#L64) fails if this is not the case. For example, I have a [repository](https://github.com/olson-sean-k/nym) that includes a Rust crate called `nym-cli` that produces a binary named `nym`. The `find` command produces no results causing downstream errors in the build.

See [this workflow log](https://github.com/olson-sean-k/nym/runs/2433749731?check_suite_focus=true) for an example of a failed run when this happens. The constructed `find` command on line 214 will fail, because the binary is named `nym` and **not** `nym-cli`:

```shell
find target/x86_64-pc-windows-gnu/release/ -maxdepth ++ 1 -type f -executable '(' -name nym-cli -o -name 'nym-cli.*' ')'
```

Perhaps a `BIN` environment variable could be introduced. If working from the root of a workspace, this could accept a list of names expected in the `target` directory, though that may not interact well with `SRC_DIR`. Thoughts?

**Comment by Douile** [comment]
Good point, I might be able to get this to work auto-magically using `cargo read-manifest` but a `BINS` variable would still be useful for specifying what binary to use.

**Comment by Douile** [comment]
Should be fixed in https://github.com/rust-build/rust-build.action/releases/tag/v1.0.2

**Comment by Douile** [comment]
Hmmm, for some reason when compiling `nym-cli/Cargo.toml` the output ends up in the parent directory

**Comment by Douile** [comment]
Okay, should have a fix :)

**Comment by Douile** [comment]
Should be fixed in 1.0.3, let me know if you have any more issue @olson-sean-k

**Comment by olson-sean-k** [comment]
The binaries are publishing successfully now! 🎉 Thanks again for taking another look. I'll go ahead and close this.

> Hmmm, for some reason when compiling nym-cli/Cargo.toml the output ends up in the parent directory

Interesting. I'm guessing that Cargo sees that `nym-cli` is part of a workspace and decides to place the build artifacts at the workspace root instead of the crate root.
